Matlab/detection/slidingWindow.m

```diff
@@ -0,0 +1,61 @@
+function [windows, boxPoint] = slidingWindow(image, winY, winX, stepY, stepX, direction)
+% SLIDINGWINDOW Splits up a image into discrete windows
+% goes through an image at a specific rate spliting it.
+% returns a matrix of the segments, and the boxpoints.
+% image --> the image to be processed. 
+% winY, winX --> the size of the cutout window.
+% stepY, stepX --> the amount of change of the starting pixel per
+% iteration default 1/3 of the size of the window in both directions.
+% direction --> 0 for horizontal 1 for vertical default 0.
+
+if nargin < 4
+    arg4 = winY/3;
+end
+
+if nargin < 5
+    arg5 = winX/3;
+end
+
+if nargin < 6
+    arg6 = 0;
+end
+
+topLeftCol = 1;
+topLeftRow = 1;
+
+% Transpose the matrix for vertical Movement.
+if direction == 1

```

Comment:
I my every pursuit of laziness and not wanting to write more code I just flip the image so the dimensions reverse allowing us to traverse it "Vertically"

Bit of a hack but works well.